### PR TITLE
fix(ext/node): add autoSelectFamily option to net.createConnection

### LINKED ISSUE #26641: ext/node: implement `autoSelectFamily` option in `net.connect`
ref: https://github.com/nodejs/node/pull/44731

This seems necessary for #25470

### PR DESCRIPTION
**Pull request by kt3k**
closes #26641 

- [x] compat test (`tests/node_compat/test/parallel/test-net-autoselectfamily.js`) is too slow in mac (like 1m30s)
- [x] compat test (`tests/node_compat/test/parallel/test-net-autoselectfamily.js`) doesn't pass on windows
- [x] `http2.connect("https://example.com")` doesn't work

### CODE REVIEW

**Comment by kt3k** [comment]
The second `createDnsServer` call seems taking 1m30s, and that seems caused because of some ops remaining even after `dnsServer.close()` call (Something looks wrong with `dgram.Socket`. I'll look into them further)

File: tests/node_compat/test/parallel/test-net-autoselectfamily.js [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+  }
+
+  if (!Array.isArray(ipv4Addrs)) {
+    ipv4Addrs = [ipv4Addrs];
+  }
+
+  // Create a DNS server which replies with a AAAA and a A record for the same host
+  const socket = dgram.createSocket('udp4');
+
+  // Note(kt3k): We use common.mustCallAtLeast instead of common.mustCall
```
Comment (satyarohith) [inline tests/node_compat/test/parallel/test-net-autoselectfamily.js:58]: nit: this should be a TODO

once both the TODOs are resolved we can remove this test from ignore in node_compat/config.jsonc

File: ext/node/polyfills/net.ts [hunk 2]
```diff
@@ -393,6 +408,107 @@ function _afterConnect(
   }
 }
 
+function _createConnectionError(req, status) {
```
Comment (satyarohith) [inline ext/node/polyfills/net.ts:411]: nit: I think we can just remove the underscore prefix for all the functions so they're easy to search in nodejs src too? or is something preventing us from doing that?

**Review by satyarohith — APPROVED** [review]
Nice work @kt3k!

LGTM

File: tests/node_compat/test/parallel/test-net-autoselectfamily.js (hunk 1 above)
Comment (kt3k) [reply tests/node_compat/test/parallel/test-net-autoselectfamily.js]: Good catch. Thanks

> once both the TODOs are resolved we can remove this test from ignore in node_compat/config.jsonc

That's correct

File: ext/node/polyfills/net.ts (hunk 2 above)
Comment (kt3k) [reply ext/node/polyfills/net.ts:411]: > I think we can just remove the underscore prefix for all the functions so they're easy to search in nodejs src too?

Sounds reasonable to me, but I would do that in another PR as there's so many of them.

Comment (kt3k) [reply ext/node/polyfills/net.ts:411]: https://github.com/denoland/deno/issues/26823

**Pull request merged**